### PR TITLE
DEPR: to_datetime('now') match Timestamp('now')

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -855,17 +851,11 @@ cdef inline bint _parse_today_now(str val, int64_t* iresult, bint utc):
...
             # GH#18705 make sure to_datetime("now") matches Timestamp("now")
-            warnings.warn(
-                "The parsing of 'now' in pd.to_datetime without `utc=True` is "
-                "deprecated. In a future version, this will match Timestamp('now') "
-                "and Timestamp.now()",
-                FutureWarning,
-                stacklevel=find_stack_level(),
-            )
-
+            iresult[0] = Timestamp("now").value
```
Comment (mroeschke) [inline pandas/_libs/tslib.pyx:858]: Looks a bit more performant to use `Timestamp.now` instead

```
In [3]: %timeit Timestamp.now()
2.04 µs ± 32.3 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [4]: %timeit Timestamp("now")
2.25 µs ± 29.1 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Comment (jbrockmendel) [reply pandas/_libs/tslib.pyx:858]: looks like `np.datetime64("now", "ns").view("i8")` is 2x faster than Timestamp.now().value

Comment (jbrockmendel) [reply pandas/_libs/tslib.pyx]: turns out Timestamp.now() was the way to go. updated.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**